### PR TITLE
HC: Refine subnet join parameter

### PR DESCRIPTION
**Pull request by sergefdrv**
## Related Issues

https://github.com/adlrocha/hc-subnet-actor/issues/2

## Proposed Changes

Introducing `JoinParams` to use in place of `Validator` in subnet `Join` method.

## Checklist

Before you mark the PR ready for review, please make sure that:
- [ ] All commits have a clear commit message.
- [ ] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] CI is green

### CODE REVIEW

File: chain/consensus/hierarchical/actors/subnet/subnet_actor.go [hunk 1]
```diff
@@ -461,6 +448,11 @@ func (st *SubnetState) addStake(rt runtime.Runtime, sourceAddr address.Address,
 		// There can only be a single miner in delegated consensus.
 		if st.Consensus != hierarchical.Delegated || len(st.Miners) < 1 {
 			st.Miners = append(st.Miners, sourceAddr)
+			st.ValidatorSet = append(st.ValidatorSet, hierarchical.Validator{
+				Subnet:  address.NewSubnetID(st.ParentID, sourceAddr),
+				Addr:    sourceAddr,
+				NetAddr: params.ValidatorNetAddr,
```
Comment (dnkolegov) [inline chain/consensus/hierarchical/actors/subnet/subnet_actor.go:454]: What if JoinParams is nil?

Comment (sergefdrv) [reply chain/consensus/hierarchical/actors/subnet/subnet_actor.go:454]: Here `params` comes from the argument of the `Join` method. I suppose the VM takes care of passing actor correct method arguments. If there is some other piece of actor's code that incorrectly passes `params == nil` then this is a programmer's error, and `addStake` will rightfully panic.

**Review by adlrocha — APPROVED** [review]

**Review by adlrocha — CHANGES_REQUESTED** [review]
review pending

**Review by dnkolegov — APPROVED** [review]
LGTM!

File: chain/consensus/hierarchical/actors/subnet/subnet_state.go [hunk 2]
```diff
@@ -59,6 +59,7 @@ type SubnetState struct {
 	MinMinerStake abi.TokenAmount
 	// List of miners in the subnet.
 	// NOTE: Consider using AMT.
+	// TODO: This field is superseded by ValidatorSet; remove it.
 	Miners []address.Address
```
Comment (adlrocha) [inline chain/consensus/hierarchical/actors/subnet/subnet_state.go:63]: This code will be removed when we have the FVM actors, we will only have to move the type definitions to its own package so we can use them to inspect the state (just as a note).

**Review by adlrocha — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**